Runtime/Scripts/BehaviorTree/RGBehaviorTreeBot.cs

```diff
@@ -0,0 +1,23 @@
+using RegressionGames.RGBotLocalRuntime;
+
+namespace RegressionGames.BehaviorTree
+{
+    public abstract class RGBehaviorTreeBot: RGUserBot
+    {
+        private BehaviorTreeNode _rootNode;
+
+        internal override void Init(long botId, string botName)
+        {
+            base.Init(botId, botName);
+            _rootNode = BuildBehaviorTree();
+        }
+
+        public override void ProcessTick(RG rgObject)
+        {
+            var result = _rootNode.Execute(rgObject);
+            RGDebug.LogVerbose($"Behavior tree completed with status: {result}");
```

Comment:
That's a good question. Returning non-`Success` from the root doesn't necessarily mean a failure. Condition nodes can return `Failure` just to indicate that the condition isn't met.

> Do we log enough information or give them any context to know which node failed and context to help determine why ?

I'll do a pass and see if I can get some per-node logging in place that makes sense. I think there's more we can do here (tracking the result of each node so we can capture the full execution flow, etc.) but I think we can iterate there as we understand the exact debugging scenarios better.

And of course, all of the user's nodes are in their project. So, they have the ability to set breakpoints in the debugger and trace execution that way.

Reply:
RE: statuses at the root node:

Failure -> For an individual node, this mean simply means the node failed - an action couldn't be completed, a condition was falsey, etc. At the root, this means that the tree couldn't successfully complete _any_ of its branches. Depending on how the user sets up their tree, a tick that resulted in a failure _could_ still effect the state in some way (ex. moving to an enemy but failing to attack it, which causes that sequence to fail).

Running -> You're waiting on some action to complete. It could be one that you started this tick, or one that you started on a previous tick. This is almost the same as "Success", but indicates that you're effectively short-circuiting the tree and returning that status immediately all the way to the root, whereas a "Success" might continue executing other nodes in the tree (in a sequence, for example)